### ISSUE TITLE
Assembler failing at overlap error adjustment

### ISSUE THREAD

**Issue by AndrewCBarnes** [marbl/canu#281]
I have a repeating problem with canu failing at the assembly point when using data from multiple smrt cells. I can assemble from a single smrt cell, but when I load fastas from 2 or 3 cells the same issue occurs. Tail of the log pasted below. Genome size estimated at 4.8m, but likely to be at least 1 plasmid. Fastas generated from bax.h5 files using dextractor, one fasta per smrt cell. Any help to troubleshoot appreciated

```
-- BEGIN ASSEMBLY
--
-- overlap error adjustment attempt 1 begins with 0 finished, and 1 to compute.
----------------------------------------
-- Starting concurrent execution on Mon Oct 24 16:12:10 2016 with 114.1 GB free disk space (1 processes; 4 concurrently)

    /Users/andrewbarnes/Photobacterium_Genome_Project/QMA0505CanuLarge/unitigging/3-overlapErrorAdjustment/oea.sh 1 > /Users/andrewbarnes/Photobacterium_Genome_Project/QMA0505CanuLarge/unitigging/3-overlapErrorAdjustment/oea.000001.out 2>&1

-- Finished on Mon Oct 24 16:12:11 2016 (1 second) with 114.1 GB free disk space
----------------------------------------
--
-- 1 overlap error adjustment jobs failed:
--   job /Users/andrewbarnes/Photobacterium_Genome_Project/QMA0505CanuLarge/unitigging/3-overlapErrorAdjustment/0001.oea FAILED.
--
-- overlap error adjustment attempt 2 begins with 0 finished, and 1 to compute.
----------------------------------------
-- Starting concurrent execution on Mon Oct 24 16:12:11 2016 with 114.1 GB free disk space (1 processes; 4 concurrently)

    /Users/andrewbarnes/Photobacterium_Genome_Project/QMA0505CanuLarge/unitigging/3-overlapErrorAdjustment/oea.sh 1 > /Users/andrewbarnes/Photobacterium_Genome_Project/QMA0505CanuLarge/unitigging/3-overlapErrorAdjustment/oea.000001.out 2>&1

-- Finished on Mon Oct 24 16:12:13 2016 (2 seconds) with 114.1 GB free disk space
----------------------------------------
--
-- 1 overlap error adjustment jobs failed:
--   job /Users/andrewbarnes/Photobacterium_Genome_Project/QMA0505CanuLarge/unitigging/3-overlapErrorAdjustment/0001.oea FAILED.
--
================================================================================
Don't panic, but a mostly harmless error occurred and canu failed.

canu failed with 'failed to adjust overlap error rates.  Made 2 attempts, jobs still failed'.
```

**Comment by skoren** [comment]
You can check the output of ```/Users/andrewbarnes/Photobacterium_Genome_Project/QMA0505CanuLarge/unitigging/3-overlapErrorAdjustment/0001.oea```
which should have more details on the error.

**Comment by AndrewCBarnes** [comment]
Thanks for the fast response. the 0001.0ea file has not been written. I think the problem lies here, but not sure how to fix it:
Initializing.
Opening gkpStore '/Users/andrewbarnes/Photobacterium_Genome_Project/QMA0505CanuLarge/unitigging/QMA0505_large.gkpStore'.
Correcting reads 1 to 7065.
Reading 53760 corrections from '/Users/andrewbarnes/Photobacterium_Genome_Project/QMA0505CanuLarge/unitigging/3-overlapErrorAdjustment/red.red'.
Correcting 151592764 bases with 42584 indel adjustments.
--Allocate 144 + 0 + 0 MB for bases, adjusts and reads.
ERROR: didn't find IDENT at Cpos=53760 for read 7065
       C[Cpos] = keep_left=0 keep_right=0 type=13 pos=25547217 readID=1633907828
Assertion failed: (C[Cpos].type == IDENT), function Correct_Frags, file overlapErrorAdjustment/correctOverlaps-Correct_Frags.C, line 296.
/Users/andrewbarnes/Photobacterium_Genome_Project/QMA0505CanuLarge/unitigging/3-overlapErrorAdjustment/oea.sh: line 50:  5714 Abort trap: 6           $bin/correctOverlaps -G /Users/andrewbarnes/Photobacterium_Genome_Project/QMA0505CanuLarge/unitigging/QMA0505_large.gkpStore -O /Users/andrewbarnes/Photobacterium_Genome_Project/QMA0505CanuLarge/unitigging/QMA0505_large.ovlStore -R $minid $maxid -e 0.039 -l 500 -c /Users/andrewbarnes/Photobacterium_Genome_Project/QMA0505CanuLarge/unitigging/3-overlapErrorAdjustment/red.red -o /Users/andrewbarnes/Photobacterium_Genome_Project/QMA0505CanuLarge/unitigging/3-overlapErrorAdjustment/$jobid.oea.WORKING

**Comment by brianwalenz** [comment]
What version of canu is this?  This sounds like a problem I fixed in the (late?) spring.

**Comment by AndrewCBarnes** [comment]
1.3. through homebrew

**Comment by brianwalenz** [comment]
Sorry for the lack of a response.  Are you still stuck here?  Did you try the latest (non-release) version of canu?

**Comment by AndrewCBarnes** [comment]
Not yet. I couldn't get canu to run on my mac from a manual install - I think it was a library issue, so I bottled and went for the homebrew route, which works well. Could be a memory issue - three smrt cells of data gives around 1000 x coverage which is a big ask for a single machine particularly at the alignment step. But these genomes are unusual with a lot of short repeats and internal self complementarity so it's a difficult assembly. Best I got out  of a single smrt cell was 5 contigs but 2 are almost certainly misassembled. It would be good to get it to finish, but it may be that a single mac with 16 gb ram doesn't have the legs for it.

Thanks for the response.
Andy

Sent from my iPad

On 21 Nov. 2016, at 7:40 pm, Brian Walenz <notifications@github.com<mailto:notifications@github.com>> wrote:


Sorry for the lack of a response. Are you still stuck here? Did you try the latest (non-release) version of canu?

-
